openmc/plotter.py

```diff
@@ -0,0 +1,588 @@
+                min_delta = np.finfo(np.float64).max
+                closest_t = -1
+                for t in data_Ts:
+                    if abs(data_Ts[t] - T) < min_delta:
+                        closest_t = t
+                sabT = "{}K".format(int(round(data_Ts[closest_t])))
+
+            # Create an energy grid composed the S(a,b) and
+            # the nuclide's grid
+            grid = nuc.energy[nucT]
+            sab_Emax = 0.
+            sab_funcs = []
+            if sab.elastic_xs:
+                elastic = sab.elastic_xs[sabT]
+                if isinstance(elastic, openmc.data.CoherentElastic):
+                    grid = np.union1d(grid, elastic.bragg_edges)
+                    if elastic.bragg_edges[-1] > sab_Emax:
+                        sab_Emax = elastic.bragg_edges[-1]
+                elif isinstance(elastic, openmc.data.Tabulated1D):
+                    grid = np.union1d(grid, elastic.x)
+                    if elastic.x[-1] > sab_Emax:
+                        sab_Emax = elastic.x[-1]
+                sab_funcs.append(elastic)
+            if sab.inelastic_xs:
+                inelastic = sab.inelastic_xs[sabT]
+                grid = np.union1d(grid, inelastic.x)
+                if inelastic.x[-1] > sab_Emax:
+                        sab_Emax = inelastic.x[-1]
+                sab_funcs.append(inelastic)
+            energy_grid = grid
+        else:
+            energy_grid = nuc.energy[nucT]
+
+        for i, mt_set in enumerate(mts):
+            # Get the reaction xs data from the nuclide
+            funcs = []
+            op = ops[i]
+            for mt, yield_check in zip(mt_set, yields[i]):
+                if mt == 2:
+                    if sab_name:
+                        # Then we need to do a piece-wise function of
+                        # The S(a,b) and non-thermal data
+                        sab_sum = openmc.data.Sum(sab_funcs)
+                        pw_funcs = openmc.data.Regions1D(
+                            [sab_sum, nuc[mt].xs[nucT]],
+                            [sab_Emax])
+                        funcs.append(pw_funcs)
+                    else:
+                        funcs.append(nuc[mt].xs[nucT])
+                elif mt in nuc:
+                    if yield_check:
+                        for prod in nuc[mt].products:
+                            if prod.particle == 'neutron' and \
+                                prod.emission_mode == 'total':
+                                func = openmc.data.Combination(
+                                    [nuc[mt].xs[nucT], prod.yield_],
+                                    [np.multiply])
+                                funcs.append(func)
+                                break
+                        else:
+                            for prod in nuc[mt].products:
+                                if prod.particle == 'neutron' and \
+                                    prod.emission_mode == 'prompt':
```

Comment:
Looks like you could easily combine this block with the previous one by changing the check on the previous one to `prod.emission_mode in ('prompt', 'total')`